lib/index-web.js

```diff
@@ -1,20 +1,33 @@
-var async         = require('async')
-var bodyParser    = require('body-parser')
-var Cookies       = require('cookies')
-var express       = require('express')
-var fs            = require('fs')
-var Handlebars    = require('handlebars')
-var renderReadme  = require('render-readme')
-var Search        = require('./search')
-var Middleware    = require('./middleware')
-var match         = Middleware.match
-var validate_name = Middleware.validate_name
-var validate_pkg  = Middleware.validate_package
+var async             = require('async')
+var bodyParser        = require('body-parser')
+var Cookies           = require('cookies')
+var express           = require('express')
+var fs                = require('fs')
+var Handlebars        = require('handlebars')
+var handlebarsLayouts = require('handlebars-layouts')
+var renderReadme      = require('render-readme')
+var Search            = require('./search')
+var Middleware        = require('./middleware')
+var Logger            = require('./logger')
+var load_plugins      = require('./plugin-loader').load_plugins
+var match             = Middleware.match
+var validate_name     = Middleware.validate_name
+var validate_pkg      = Middleware.validate_package
 
 module.exports = function(config, auth, storage) {
   var app = express.Router()
   var can = Middleware.allow(auth)
 
+  var logger = Logger.logger.child({ sub: 'web' })
+
+  var plugin_params = {
+    config: config,
+    logger: logger,
+    auth: auth,
+    storage: storage,
```

Comment:
By the way, for the `auth` part, I am already using it in
https://github.com/tuvistavie/sinopia-users-management
For the moment, this allows me to add users easily through the web interface.